runhouse/resources/hardware/cluster.py

```diff
@@ -128,36 +128,40 @@ def address(self, addr):
 
     @property
     def client(self):
-        if not self._http_client:
-            if not self._ping(retry=True):
-                # ping cluster, and refresh ips if ondemand cluster and first ping fails
-                raise Exception(
-                    f"Could not reach cluster {self.name} ({self.ips}). Is it up?"
-                )
-
+        def check_connect_server():
             connect_call = threading.Thread(
                 target=self.connect_server_client, kwargs={"force_reconnect": True}
             )
             connect_call.start()
             connect_call.join(timeout=5)
             if connect_call.is_alive():
+                return False
+            return True
+
+        if not check_connect_server():
+            if self.__class__.__name__ == "OnDemandCluster":
+                self._update_from_sky_status(dryrun=False)
```

Comment:
Is this called every time we first form the connection, or only if it fails based on the IP in the sky db without refreshing?

Reply:
we try to form the connection, and if it fails (5 sec timeout) then we do a sky refresh + ping to check if the cluster is available before trying to connect again